File: src/test/java/no/novari/configuration/validation/valueparsability/validators/BooleanParsabilityValidatorTest.java
package no.novari.configuration.validation.valueparsability.validators;

import no.novari.configuration.model.configuration.entities.ValueMapping;
import no.novari.configuration.validation.valueparsability.validators.BooleanParsabilityValidator;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class BooleanParsabilityValidatorTest {

    private BooleanParsabilityValidator booleanParsabilityValidator;


    @BeforeEach
    void setup() {
        booleanParsabilityValidator = new BooleanParsabilityValidator();
    }

    @Test
    void getTypeToValidate() {
        assertEquals(ValueMapping.Type.BOOLEAN, booleanParsabilityValidator.getTypeToValidate());
    }

    @Test
    void shouldReturnTrueIfValueIsTrue() {
        boolean valid = booleanParsabilityValidator.isValid("true");
        assertTrue(valid);
    }

    @Test
    void shouldReturnTrueIfValueIsFalse() {
        boolean valid = booleanParsabilityValidator.isValid("false");
        assertTrue(valid);
    }

    @Test
    void shouldReturnFalseIfValueIsEmpty() {
        boolean valid = booleanParsabilityValidator.isValid("");
        assertFalse(valid);
    }

    @Test
    void shouldReturnFalseIfValueIsBlank() {
        boolean valid = booleanParsabilityValidator.isValid(" ");
        assertFalse(valid);
    }

    @Test
    void shouldReturnFalseIfValueIsTruest() {
        boolean valid = booleanParsabilityValidator.isValid("truest");
        assertFalse(valid);
    }
}

File: src/test/kotlin/no/novari/flyt/configuration/validation/valueparsability/validators/BooleanParsabilityValidatorTest.kt
package no.novari.flyt.configuration.validation.valueparsability.validators

import no.novari.flyt.configuration.model.configuration.entities.ValueMapping
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class BooleanParsabilityValidatorTest {
    private lateinit var booleanParsabilityValidator: BooleanParsabilityValidator

    @BeforeEach
    fun setUp() {
        booleanParsabilityValidator = BooleanParsabilityValidator()
    }

    @Test
    fun `reports the value type it validates`() {
        assertEquals(ValueMapping.Type.BOOLEAN, booleanParsabilityValidator.getTypeToValidate())
    }

    @Test
    fun `accepts the value true`() {
        assertTrue(booleanParsabilityValidator.isValid("true"))
    }

    @Test
    fun `accepts the value false`() {
        assertTrue(booleanParsabilityValidator.isValid("false"))
    }

    @Test
    fun `rejects an empty value`() {
        assertFalse(booleanParsabilityValidator.isValid(""))
    }

    @Test
    fun `rejects a blank value`() {
        assertFalse(booleanParsabilityValidator.isValid(" "))
    }

    @Test
    fun `rejects a value that only looks like a boolean`() {
        assertFalse(booleanParsabilityValidator.isValid("truest"))
    }
}
